File: queue/queue.js
function Queue() {
    this.queue_list = [];
}

Queue.prototype.enqueue = function(data) {
    this.queue_list.splice(0, 0, data);
}

Queue.prototype.dequeue = function() {
    return this.queue_list.pop();
}

Queue.prototype.isEmpty = function() {
    return this.queue_list.length === 0;
}

Queue.prototype.size = function() {
    return this.queue_list.length;
}

//TESTING FUNCTIONS

function print(data) {
    console.log(data);
}

print("Initializing Queue");
qu = new Queue();
print("Enqueueing 'Hello'");
qu.enqueue("Hello");
print( "isEmpty: " + qu.isEmpty() );
print("Enqueueing 'World'");
qu.enqueue("World");
print( "Size: " + qu.size() );
print( "Dequeue: " + qu.dequeue() );
print( "Dequeue: " + qu.dequeue() );
print( "isEmpty: " + qu.isEmpty() );

File: queue/queue.py
class Queue:
    def __init__(self):
        self.queue_list = []

    def enqueue(self, data):
        self.queue_list.insert(0, data)

    def dequeue(self):
        return self.queue_list.pop()

    def isEmpty(self):
        return self.queue_list == []

    def size(self):
        return len(self.queue_list)